platform/commonUI/browse/src/creation/CreateAction.js

```diff
@@ -88,22 +85,26 @@ define(
         CreateAction.prototype.perform = function () {
             var newModel = this.type.getInitialModel(),
                 parentObject = this.navigationService.getNavigation(),
-                newObject,
-                editableObject;
+                editorCapability,
+                newObject;
 
             newModel.type = this.type.getKey();
             newObject = parentObject.useCapability('instantiation', newModel);
-            editableObject = new EditableDomainObject(newObject, this.$q);
-            editableObject.setOriginalObject(parentObject);
-            editableObject.getCapability('status').set('editing', true);
-            editableObject.useCapability('mutation', function(model){
+            newObject.useCapability('mutation', function(model){
                 model.location = parentObject.getId();
             });
+            editorCapability = newObject.getCapability("editor");
 
-            if (countEditableViews(editableObject) > 0 && editableObject.hasCapability('composition')) {
-                this.navigationService.setNavigation(editableObject);
+            if (countEditableViews(newObject) > 0 && newObject.hasCapability('composition')) {
```

Comment:
In the UI, we have an Edit button that either switches to Edit mode or shows an Edit Properties dialog. IIRC these are two different `edit` actions with same/similar policies. (With similar logic to this if.)

Are there tweaks to those `edit` actions that would allow us to replace this `if-else` with a single `perform('edit')`? (e.g. if they both return promises for success or failure...)

That said, it's a bit out of scope for this issue (is there another refactoring issue where this would be appropriate?)